Index: sites/all/modules/contrib/itweak_upload/itweak_upload.module
===================================================================
--- sites/all/modules/contrib/itweak_upload/itweak_upload.module	(revision 4354)
+++ sites/all/modules/contrib/itweak_upload/itweak_upload.module	(working copy)
@@ -10,7 +10,7 @@
  * TODO LIST (FIXME)
  *
  * GOODENOUGH feature: create imagecache presets:
- * DONE 1. upon our install 
+ * DONE 1. upon our install
  * DONE 2. recommend imagecache if not installed (where?)
  * NO   3. hook to imagecache install to create presets when imagecache is installed after us.
  * - translations .pot
@@ -29,13 +29,6 @@
  */
 function itweak_upload_init() {
   drupal_add_css(drupal_get_path('module', 'itweak_upload') . '/itweak_upload.css');
-  if (module_exists('jcarousel')) {
-    $selector = '.attachment-thumbs';
-	$options = array('scroll' => 1);
-	$skin = 0 ? 'ie7' : 'tango';
-	$skin_path = '';
-    jcarousel_add($selector, $options, $skin, $skin_path);
-  }
 }
 
 /**
@@ -78,19 +71,19 @@
   $presets = array();
   
   $presets['AttachmentThumbnail'] = array(
-    'presetname' => 'AttachmentThumbnail', 
+    'presetname' => 'AttachmentThumbnail',
     'actions' => array(
       0 => array(
-        'weight' => '0', 
-        'module' => 'imagecache', 
-        'action' => 'imagecache_scale_and_crop', 
+        'weight' => '0',
+        'module' => 'imagecache',
+        'action' => 'imagecache_scale_and_crop',
         'data' => array(
-          'width' => 60, 
-          'height' => 60, 
-          'upscale' => 1 
-        ) 
-      ) 
-    ) 
+          'width' => 60,
+          'height' => 60,
+          'upscale' => 1
+        )
+      )
+    )
   );
   
   return $presets;
@@ -182,8 +175,8 @@
  *  Node type or 'comment'.
  */
 function _itweak_upload_upload_form_alter(&$form, $form_state, $type) {
-  $node_type_name = ($type == 'comment') 
-    ? t('comment') 
+  $node_type_name = ($type == 'comment')
+    ? t('comment')
     : strtolower(node_get_types('name', $form['#node']->type));
   drupal_add_js(drupal_get_path('module', 'itweak_upload') . '/itweak_upload.js');
   $collapse = variable_get('itweak_upload_collapse_' . $type, 0);
@@ -198,7 +191,7 @@
 
 /*
  * Get a hierarchical setting.
- * Order of settings is: 
+ * Order of settings is:
  *   node type specific > node type default > global default > given default
  * When any of the settings is '_default', the preceding one takes place.
  * @param $setting_base
@@ -266,15 +259,15 @@
   $handler = '';
   if (module_exists('lightbox2')) {
     switch ($link_option) {
-      case 'lightbox2': 
+      case 'lightbox2':
         $handler = 'lightbox';
         $options['attributes'] = array('rel' => $handler);
         break;
-      case 'lightbox2grouped': 
+      case 'lightbox2grouped':
         $handler = 'lightbox' . ($group ? '[attachment-thumb-' . $group . ']' : '[attachment-thumb]');
         $options['attributes'] = array('rel' => $handler);
         break;
-      case 'lightbox2slideshow': 
+      case 'lightbox2slideshow':
         $handler = 'lightshow' . ($group ? '[attachment-thumb-' . $group . ']' : '[attachment-thumb]');
         $options['attributes'] = array('rel' => $handler);
         break;
@@ -412,7 +405,7 @@
 
     case 'node_type_form':
       module_load_include('admin.inc', 'itweak_upload');
-      _itweak_upload_node_type_form($form);  
+      _itweak_upload_node_type_form($form);
       break;
     case 'upload_admin_settings':
       module_load_include('admin.inc', 'itweak_upload');
@@ -473,7 +466,7 @@
 
 /**
  * Implementation of theme_upload_form_new().
- * Theme the fieldset for new attachment. 
+ * Theme the fieldset for new attachment.
  */
 function itweak_upload_upload_form_new($form) {
   unset($form['new']['upload']['#title']);
@@ -587,7 +580,7 @@
     }
   }
   if (count($rows)) {
-    return 
+    return
       '<div class="itu-attachments">'
       . theme('table', $header, $rows, array('class' => 'itu-attachment-list', 'id' => 'attachments'))
       .'</div>'
@@ -702,7 +695,7 @@
 function itweak_upload_nodeapi(&$node, $op, $teaser, $page) {
   switch ($op) {
     case 'view':
-      $files_display = $teaser 
+      $files_display = $teaser
         ? variable_get('itweak_upload_teaser_display_' . $node->type, 0)
         : variable_get('itweak_upload_node_display_' . $node->type, 1);
       if ($files_display && $node->files && $node->nid && user_access('view uploaded files')) {
@@ -892,7 +885,7 @@
  *  Operation
  */
 function itweak_upload_comment(&$comment, $op) {
-  if ($op != 'view' || !module_exists('comment_upload') 
+  if ($op != 'view' || !module_exists('comment_upload')
     || !user_access('view files uploaded to comments')) {
     return;
   }
