Index: biblio.module
===================================================================
--- biblio.module	(revision 4888)
+++ biblio.module	(working copy)
@@ -2381,8 +2381,8 @@
 }
 
 function biblio_get_title_url_info($node, $base = 'biblio', $inline = FALSE) {
-  $view_mode = ($inline) ? 'viewinline' : 'view';
-  $path = $base .'/'. $view_mode;
+  
+  $path = ($inline)? $base .'/'. $view_mode : "node";
 
   return array('link' =>  ((variable_get('biblio_link_title_url', 0) && !empty($node->biblio_url)) ? $node->biblio_url : "$path/$node->nid" ),
                'options' =>
