Index: biblio/biblio.module
===================================================================
--- biblio/biblio.module	(revision 2170)
+++ biblio/biblio.module	(working copy)
@@ -375,7 +375,6 @@
     case 'update':
       if (user_access('edit all biblio entries')) return TRUE;
       if (user_access('edit own biblio entries') && $user->uid == $node->uid) return TRUE;
-      if (user_access('edit own biblio entries') && $user->uid != $node->uid) return FALSE;
       break;
     case 'view':
       if ((variable_get('biblio_view_only_own', 0)) && $user->uid != $node->uid) return FALSE;
