Index: apachesolr_attachments.admin.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/apachesolr_attachments/apachesolr_attachments.admin.inc,v
retrieving revision 1.6
diff -u -p -r1.6 apachesolr_attachments.admin.inc
--- apachesolr_attachments.admin.inc	3 Feb 2010 18:34:22 -0000	1.6
+++ apachesolr_attachments.admin.inc	10 Jun 2010 13:56:45 -0000
@@ -274,10 +280,23 @@ function apachesolr_attachments_add_docu
       $document->ss_file_node_title = apachesolr_clean_text($node->title);
       $document->ss_file_node_url = url('node/' . $node->nid, array('absolute' => TRUE));
 
+      if (module_exists('apachesolr_og')) {
+        // Index group posts
+        if (!empty($node->og_groups)) {
+          foreach ($node->og_groups as $gid) {
+            $document->setMultiValue(apachesolr_og_gid_key(), $gid);
+          }
+        }
+        elseif (isset($node->og_description) && variable_get('apachesolr_og_groupnode', 0)) {
+          // Index the group node itself as in the group.
+          $document->setMultiValue(apachesolr_og_gid_key(), $node->nid);
+        }
+      }
+  
       apachesolr_add_taxonomy_to_document($document, $node);
 
       if (module_exists('apachesolr_nodeaccess')) {
-        apachesolr_nodeaccess_apachesolr_update_index($document, $node);
+        apachesolr_nodeaccess_apachesolr_update_index($document, $node, $namespace);
       }
       drupal_alter('apachesolr_attachment_index', $document, $node, $file);
 