? .svn
? enable_multiple_themes.patch
? contrib/.svn
? core/.svn
Index: core/system.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/install_profile_api/core/Attic/system.inc,v
retrieving revision 1.1.2.4.2.1
diff -u -p -r1.1.2.4.2.1 system.inc
--- core/system.inc	6 Apr 2009 17:32:47 -0000	1.1.2.4.2.1
+++ core/system.inc	10 Jul 2009 13:48:57 -0000
@@ -13,10 +13,13 @@
  * @param $theme
  *   Unique string that is the name of theme.
  */
-function install_enable_theme($theme) {
+function install_enable_theme($themes) {
   system_theme_data();
-  db_query("UPDATE {system} SET status = 1 WHERE type = 'theme' and name = '%s'", $theme);
-  system_initialize_theme_blocks($theme);
+  $themes = is_array($themes) ? $themes : array($themes);
+  foreach($themes as $theme){
+    db_query("UPDATE {system} SET status = 1 WHERE type = 'theme' and name = '%s'", $theme);
+    system_initialize_theme_blocks($theme);
+  }
 }
 
 /**
