Index: itweak_upload.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/itweak_upload/itweak_upload.module,v
retrieving revision 1.1.2.19
diff -u -p -r1.1.2.19 itweak_upload.module
--- itweak_upload.module	8 Feb 2010 09:33:54 -0000	1.1.2.19
+++ itweak_upload.module	28 Jun 2010 19:28:06 -0000
@@ -714,7 +714,9 @@ $show_caption = FALSE;   // FIXME - sett
         $teaser_images_max = variable_get('itweak_upload_teaser_images_' . $node->type, 0);
         if ($teaser_images_max === '') $teaser_images_max = -1;
         $cnt_other = _itweak_upload_preprocess_files($node->files, $thumbnails, $files_display, $derivative, $show_title, $show_caption, $options);
-
+        
+        $node->itweak_upload = $thumbnails;
+        
         // Add regular attachment list
         if ($cnt_other) {
           if ($files_display != 4) {
